shared_model/backend/protobuf/queries/proto_query.hpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_QUERY_HPP
+#define IROHA_SHARED_MODEL_PROTO_QUERY_HPP
+
+#include <boost/range/numeric.hpp>
+
+#include "backend/protobuf/common_objects/trivial_proto.hpp"
+#include "backend/protobuf/common_objects/signature.hpp"
+#include "backend/protobuf/queries/proto_get_account.hpp"
+#include "interfaces/queries/query.hpp"
+#include "queries.pb.h"
+#include "utils/lazy_initializer.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T, typename Archive>
+auto load_query(Archive &&ar) {
+  int which = ar.payload()
+                  .GetDescriptor()
+                  ->FindOneofByName("query")
+                  ->field(ar.payload().query_case())
+                  ->index_in_oneof();
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::Query::QueryVariantType>(
+      std::forward<Archive>(ar), which);
+}
+
+namespace shared_model {
+  namespace proto {
+    class Query final : public CopyableProto<interface::Query,
+                                             iroha::protocol::Query,
+                                             Query> {
+     private:
+      /// polymorphic wrapper type shortcut
+      template <typename Value>
+      using w = detail::PolymorphicWrapper<Value>;
+
+      /// lazy variant shortcut
+      template <typename T>
+      using Lazy = detail::LazyInitializer<T>;
+
+      using LazyVariantType = Lazy<QueryVariantType>;
+
+     public:
+      /// type of proto variant
+      using ProtoQueryVariantType = boost::variant<w<GetAccount>>;
+
+      /// list of types in proto variant
+      using ProtoQueryListType = ProtoQueryVariantType::types;
+
+      template <typename QueryType>
+      explicit Query(QueryType &&query)
+          : CopyableProto(std::forward<QueryType>(query)),
+            variant_(
+                [this] { return load_query<ProtoQueryListType>(*proto_); }),
+            blob_([this] { return BlobType(proto_->SerializeAsString()); }),
+            signatures_([this] {
+              SignatureSetType set;
+              SignatureType sig(new Signature(proto_->signature()));
+              set.emplace(sig);
```

Comment:
Either use insert, or call with the `Type(proto_->signature())`

Reply:
The constructor accepts only pointers